pymeasure/instruments/agilent/agilent8648.py

```diff
@@ -0,0 +1,287 @@
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True
+    )
+
+    am_depth = Instrument.control(
+        "AM:DEPT?", "AM:DEPT %g",
+        "Control AM depth in %.",
+        validator=strict_range,
+        values=[0, 99.9],
+    )
+
+    am_enabled = Instrument.control(
+        "AM:STAT?", "AM:STAT %g",
+        "Control AM enable.",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True
+    )
+
+    am_source = Instrument.control(
+        "AM:SOUR?", "AM:SOUR %s",
+        "Control AM source (str).",
+        validator=strict_discrete_set,
+        values={"INT", "INT2", "EXT", "INT,EXT"}
+    )
+
+    am_int_frequency = Instrument.control(
+        "AM:INT:FREQ?", "AM:INT:FREQ %gHZ",
+        "Control INT frequency.",
+        validator=strict_discrete_set,
+        values={"1000", "400"}
+    )
+
+    am_ext_coupling = Instrument.control(
+        "AM:EXT:COUP?", "AM:EXT:COUP %s",
+        "Control external coupling for AM (str).",
+        validator=strict_discrete_set,
+        values={"DC", "AC"}
+    )
+
+    fm_deviation = Instrument.control(
+        "FM:DEV?", "FM:DEV %gHZ",
+        "Control FM deviation in Hz.",
+        validator=strict_range,
+        values=[10000, 99900],
+    )
+
+    fm_enabled = Instrument.control(
+        "FM:STAT?", "FM:STAT %g",
+        "Control FM enable.",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True
+    )
+
+    fm_source = Instrument.control(
+        "FM:SOUR?", "FM:SOUR %s",
+        "Control FM source (str).",
+        validator=strict_discrete_set,
+        values={"INT", "INT2", "EXT", "INT,EXT"}
+    )
+
+    fm_int_frequency = Instrument.control(
+        "FM:INT:FREQ?", "FM:INT:FREQ %gHZ",
+        "Control INT frequency.",
+        validator=strict_discrete_set,
+        values={"1000", "400"}
+    )
+
+    fm_ext_coupling = Instrument.control(
+        "FM:EXT:COUP?", "FM:EXT:COUP %s",
+        "Control external coupling for FM (str).",
+        validator=strict_discrete_set,
+        values={"DC", "AC"}
+    )
+
+    pm_deviation = Instrument.control(
+        "PM:DEV?", "PM:DEV %gRAD",
+        "Control PM deviation in radian.",
+        validator=strict_range,
+        values=[0, 10],
+    )
+
+    fm_enabled = Instrument.control(
+        "PM:STAT?", "PM:STAT %g",
+        "Control PM enable.",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True
+    )
+
+    pm_source = Instrument.control(
+        "PM:SOUR?", "PM:SOUR %s",
+        "Control PM source (str).",
+        validator=strict_discrete_set,
+        values={"INT", "INT2", "EXT", "INT,EXT"}
+    )
+
+    pm_int_frequency = Instrument.control(
+        "PM:INT:FREQ?", "PM:INT:FREQ %gHZ",
+        "Control INT frequency.",
+        validator=strict_discrete_set,
+        values={"1000", "400"}
+    )
+
+    pm_ext_coupling = Instrument.control(
```

Comment:
if you make it  (for example) `pm_ext_ac_coupling_enabled`, you can make it a boolean instead of a string property.

Reply:
Do you think it will be more convenient?

Reply:
booleans are more convenient: `pm_ext_ac_coupling_enabled = True` is self-explanatory and you cannot get a typo in the string.